internal/registry/registry.go

```diff
@@ -113,7 +123,48 @@ func (r *registry) ExtractToolkitRelease(layer v1.Layer) (*DriverToolkitEntry, e
 	return dtk, nil
 }
 
-func (r *registry) getPullOptions(ctx context.Context, image string, po *ootov1alpha1.PullOptions) (*RepoPullConfig, error) {
+func (r *registry) getAuthForRegistry(ctx context.Context, registry string, ps *corev1.LocalObjectReference, psNamespace string) (authn.Authenticator, error) {
+
+	secret := corev1.Secret{}
+	secretNamespacedName := types.NamespacedName{
+		Name:      ps.Name,
+		Namespace: psNamespace,
+	}
+	if err := r.client.Get(ctx, secretNamespacedName, &secret); err != nil {
+		return nil, fmt.Errorf("cannot find secret %v: %w", secretNamespacedName, err)
+	}
+
+	authData, ok := secret.Data[".dockerconfigjson"]
```

Comment:
I think `crane` features helpers that might do all that work for us. Please have a look at https://github.com/qbarrand/oot-operator/pull/91/files#diff-896098bd06b79a03757341c930e0c577c9457ee3acd25751642a0e95782e96bfR75.

Reply:
Fixed.